src/Repository/PetRepository.php

```diff
@@ -18,4 +18,11 @@ public function __construct(RegistryInterface $registry)
     {
         parent::__construct($registry, Pet::class);
     }
+
+    public function persist(Pet $pet): void
```

Comment:
This can probably be extracted into a trait, since you have it in both repositories